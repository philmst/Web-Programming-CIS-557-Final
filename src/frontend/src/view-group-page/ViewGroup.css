.info-area {
  height: 40%;
  width: 70vw;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  gap: 50px;
}

.group-view {
  display: flex;
  background-color: #F3EFF5;

  font-family: "LibertinusSans";
  transition: all 0.2s ease;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 100%;
  padding: 0px;
}

.view-area {
  border-top: 0.5px solid rgb(10, 17, 40);
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.posts {
  border-top: 0.5px solid rgb(10, 17, 40);
  height: 60%;
  width: 70vw;

  gap: 20px 20px;
}

/* post input */
.text-input {
  height: 100px;
  border: 0.5px solid black;
  border-radius: 5px;
  
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

input[type="post"] {
  width: 400px;
  height: 50px;
}

input[type="reply"] {
  width: 230px;
  height: 50px;
}

#revokeAdmin, #addAdmin {
  width: 200px;
}

#submit {
  background-color: rgb(18,130,162);
  color: white;
}

/* posts style */

.post-container {
  background-color: #F3EFF5;
  border: 2px solid #454955;
  border-radius: 6px;
  padding: 15px;

  gap: 20px;
  box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.25);
  font-family: "LibertinusSans";
  transition: all 0.2s ease;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 100px;
}

.posts-area {
  height: 100px;
  border: 0.5px solid black;
  border-radius: 5px;
}

.reply {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.reply-input {
  width: 200px;
  height: 75px;
}

div[type="flag-and-delete"] {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  gap: 20px;
}

button[type="delete"] {
  width: 50px;
  border-radius: 5px;
  background-color: #87BCDE;
}

button[type="flag"] {
  width: 50px;
  border-radius: 5px;
  background-color: #87BCDE;
}

button[type="reply-button"] {
  width: 100px;
  border-radius: 5px;
  background-color: #87BCDE;
}

/* replies style */

.reply-container {
  background-color: #F3EFF5;
  border: 2px solid #454955;
  border-radius: 6px;
  padding: 15px;

  gap: 20px;
  box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.25);
  font-family: "LibertinusSans";
  transition: all 0.2s ease;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  width: 300px;
  height: 50px;
}