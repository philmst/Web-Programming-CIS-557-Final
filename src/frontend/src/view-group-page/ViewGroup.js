import { useState, useEffect, React } from 'react';
import './ViewGroup.css';

// eslint-disable-next-line
const lib = require('./ViewGroupModule');

function ViewGroup(props) {
  // eslint-disable-next-line
  const { changeState, state } = props;

  const [groupAndAdmins, setGroupAndAdmins] = useState({ group: {}, admins: [] });
  const updateState = async () => {
    const group = await lib.getGroup(state.viewingGroup);
    const admins = await lib.getAdmins(state.viewingGroup);

    setGroupAndAdmins(() => ({
      group,
      admins,
    }));
  };

  const processAdmins = (admins) => {
    let adminLst = '';
    admins.forEach((admin) => {
      const toShow = (admin.is_creator === 1 ? `${admin.user_name} (creator)` : `${admin.user_name}`);

      adminLst += (adminLst === '' ? `${toShow} ` : `, ${toShow}`);
    });
    return adminLst;
  };

  // eslint-disable-next-line no-unused-vars
  const [allPosts, setAllPosts] = useState([]);

  const updatePosts = async () => {
    const posts = await lib.getPosts(changeState, state.viewingGroup);

    setAllPosts(posts.result);
    lib.parsePosts(posts.result[0]);
  };

  const createPost = async () => {
    // eslint-disable-next-line
    const res = await lib.createPost(changeState,
      state.viewingGroup,
      state.userId,
      document.getElementById('post').value);

    const posts = await lib.getPosts(changeState, state.viewingGroup);
    // eslint-disable-next-line no-console
    console.log(`posts: ${JSON.stringify(posts.result[0])}`);

    setAllPosts(posts.result[0]);

    // eslint-disable-next-line no-console
    console.log(`all posts: ${JSON.stringify(allPosts)}`);

    changeState({ link: '/viewgroup' });
    /*
    lib.parsePosts(posts.result[0]);
    */
  };

  // eslint-disable-next-line no-unused-vars
  const [allReplies, setAllReplies] = useState([]);

  // eslint-disable-next-line no-unused-vars
  const updateReplies = async () => {
    const posts = await lib.getPosts(changeState, state.viewingGroup);
    const replies = await lib.getReplies(changeState, state.viewingGroup);

    setAllReplies(replies.result);

    if (replies.result) {
      lib.parseReplies(posts.result[0], replies.result[0]);
      lib.parseOnclicks(state, changeState, posts.result[0], replies.result[0]);
    }
  };

  const updates = async () => {
    await updateState();
    await updatePosts();
    await updateReplies();
  };

  useEffect(() => { updates(); }, [state]);

  return (
    <div className="container">
      <div className="header">
        <div className="social-media-title">Social Media</div>
        <div className="profile-picture">
          <div className="img" />
        </div>
        <div className="username">Hi, username!</div>
      </div>

      <div className="top-navbar">
        <div className="home-link">Home</div>
        <div className="profile-link">Profile</div>
      </div>

      <div className="main-container">

        <div className="side-navbar">
          <button type="submit" className="notifications" onClick={() => changeState({ link: '/notifications' })}>Notifications</button>
          <button type="submit" className="events">Events</button>
          <button type="submit" className="groups" onClick={() => changeState({ link: '/groups' })}>Groups</button>
          <button type="submit" className="videos">Videos</button>
          <button type="submit" className="photos">Photos</button>
        </div>

        <div className="main-area">
          <div className="info-area">
            <div className="group-view">
              <div className="group-info">
                <ul>
                  <li id="group-id">
                    {`Group Id: ${groupAndAdmins.group.group_id}`}
                  </li>
                  <li id="group-name">
                    {`Group Name: ${groupAndAdmins.group.group_name}`}
                  </li>
                  <li id="group-description">
                    {`Group Description: ${groupAndAdmins.group.group_description}`}
                  </li>
                  <li id="group-admins">
                    {`Admins: ${processAdmins(groupAndAdmins.admins)}`}
                  </li>
                  <li id="is-public">
                    {(groupAndAdmins.group.is_public === 1 ? 'Public' : 'Private')}
                  </li>
                </ul>
              </div>
            </div>
          </div>

          <div className="view-area" id="groups-area">
            <div id="revoke">
              { state.link.includes('error') ? (<p>Error: cannot revoke creator of group</p>) : (null)}
              <label htmlFor="revokeAdmin">
                {'Revoke Admin: '}
                <input id="revokeAdmin" type="text" placeholder="Enter username" />
              </label>
              <input type="button" value="Submit" id="submit" onClick={() => lib.revokeAdmin(state, changeState, groupAndAdmins)} />
            </div>
            <div id="add">
              <label htmlFor="addAdmin">
                {'Add Admin: '}
                <input id="addAdmin" type="text" placeholder="Enter username" />
              </label>
              <input type="button" value="Submit" id="submit" onClick={() => lib.addAdmin(groupAndAdmins)} />
            </div>
          </div>

          <div className="posts-area">
            <div className="text-input">
              Message:
              <input className="post-input" id="post" type="post" placeholder="Post message to group" />
              <button className="post-button" type="submit" onClick={() => createPost()}> Post Message </button>
            </div>
            <div className="posts" id="posts-area" />
          </div>
        </div>

        <div className="message-updates">
          Message updates
        </div>

      </div>
    </div>
  );
}

export default ViewGroup;
