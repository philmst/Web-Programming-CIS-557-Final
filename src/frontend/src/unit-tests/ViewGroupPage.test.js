import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ViewGroup from '../view-group-page/ViewGroup';

/**
 * @jest-environment jsdom
 */

const changeLink = jest.fn();

describe('Test View Group Page UI', () => {
  test('View group page renders correctly', () => {
    const component = renderer.create(<ViewGroup changeState={changeLink} state={{username: 'me', link: '/viewgroup', viewingGroup: 0}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
    
  test('View group page error renders correctly', () => {
    const component = renderer.create(<ViewGroup changeState={changeLink} state={{username: 'me', link: '/viewgroup/error', viewingGroup: 0}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
    
  test('View group page post error renders correctly', () => {
    const component = renderer.create(<ViewGroup changeState={changeLink} state={{username: 'me', link: '/viewgroup/post/error', viewingGroup: 0}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
    
  test('View group page admin renders correctly', () => {
    const component = renderer.create(<ViewGroup changeState={changeLink} state={{username: 'me', link: '/viewgroup/admin', viewingGroup: 0}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});