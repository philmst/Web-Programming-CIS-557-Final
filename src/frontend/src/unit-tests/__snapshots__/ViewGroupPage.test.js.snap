// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test View Group Page UI View group page admin renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, me!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        onClick={[Function]}
        type="submit"
      >
        Analytics
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="invitations"
        onClick={[Function]}
        type="submit"
      >
        Group Invitations and Requests 
      </button>
      <button
        className="photos"
        type="submit"
      >
        Photos
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="info-area"
        id="info-area"
      >
        <div
          className="group-view"
        >
          <div
            className="group-info"
          >
            <ul>
              <li
                id="group-id"
              >
                Group Id: undefined
              </li>
              <li
                id="group-name"
              >
                Group Name: undefined
              </li>
              <li
                id="group-description"
              >
                Group Description: undefined
              </li>
              <li
                id="group-admins"
              >
                Admins: 
              </li>
              <li
                id="is-public"
              >
                Private
              </li>
            </ul>
          </div>
        </div>
        <div
          className="join-group"
          id="join-group-id"
        >
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Leave group"
          />
        </div>
      </div>
       
      <div
        className="view-area"
        id="groups-area"
      >
        <div
          id="revoke"
        >
          <label
            htmlFor="revokeAdmin"
          >
            Revoke Admin: 
            <input
              id="revokeAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add"
        >
          <p>
            Error: potential admin is not a group member
          </p>
          <label
            htmlFor="addAdmin"
          >
            Add Admin: 
            <input
              id="addAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add-non-admin"
        >
          <label
            htmlFor="addNonAdmin"
          >
            Invite a non-Admin member:
            <input
              id="addNonAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
      </div>
      <div
        className="posts-area"
      >
        <div
          className="text-input"
        >
          Create Post:
          <div
            id="options"
          >
            <label
              className="list"
              htmlFor="textInput"
            >
              <input
                id="textInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Text message
            </label>
            <label
              className="list"
              htmlFor="audioInput"
            >
              <input
                id="audioInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Audio message
            </label>
            <label
              className="list"
              htmlFor="videoInput"
            >
              <input
                id="videoInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Video message
            </label>
            <label
              className="list"
              htmlFor="imageInput"
            >
              <input
                id="imageInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Image message
            </label>
          </div>
          <div
            id="form"
          >
            <textarea
              id="postCaption"
              placeholder="Post message to group"
            />
          </div>
          <button
            className="post-button"
            onClick={[Function]}
            type="submit"
          >
             Post Message 
          </button>
        </div>
        <div
          className="posts"
          id="posts-area"
        />
      </div>
    </div>
    <div
      className="side-navbar"
      id="forMessages"
    >
      <button
        className="messages"
        onClick={[Function]}
        type="submit"
      >
        Messages
      </button>
    </div>
  </div>
</div>
`;

exports[`Test View Group Page UI View group page error renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, me!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        onClick={[Function]}
        type="submit"
      >
        Analytics
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="invitations"
        onClick={[Function]}
        type="submit"
      >
        Group Invitations and Requests 
      </button>
      <button
        className="photos"
        type="submit"
      >
        Photos
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="info-area"
        id="info-area"
      >
        <div
          className="group-view"
        >
          <div
            className="group-info"
          >
            <ul>
              <li
                id="group-id"
              >
                Group Id: undefined
              </li>
              <li
                id="group-name"
              >
                Group Name: undefined
              </li>
              <li
                id="group-description"
              >
                Group Description: undefined
              </li>
              <li
                id="group-admins"
              >
                Admins: 
              </li>
              <li
                id="is-public"
              >
                Private
              </li>
            </ul>
          </div>
        </div>
        <div
          className="join-group"
          id="join-group-id"
        >
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Leave group"
          />
        </div>
      </div>
       
      <div
        className="view-area"
        id="groups-area"
      >
        <div
          id="revoke"
        >
          <p>
            Error: cannot revoke creator of group
          </p>
          <label
            htmlFor="revokeAdmin"
          >
            Revoke Admin: 
            <input
              id="revokeAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add"
        >
          <label
            htmlFor="addAdmin"
          >
            Add Admin: 
            <input
              id="addAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add-non-admin"
        >
          <label
            htmlFor="addNonAdmin"
          >
            Invite a non-Admin member:
            <input
              id="addNonAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
      </div>
      <div
        className="posts-area"
      >
        <div
          className="text-input"
        >
          Create Post:
          <div
            id="options"
          >
            <label
              className="list"
              htmlFor="textInput"
            >
              <input
                id="textInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Text message
            </label>
            <label
              className="list"
              htmlFor="audioInput"
            >
              <input
                id="audioInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Audio message
            </label>
            <label
              className="list"
              htmlFor="videoInput"
            >
              <input
                id="videoInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Video message
            </label>
            <label
              className="list"
              htmlFor="imageInput"
            >
              <input
                id="imageInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Image message
            </label>
          </div>
          <div
            id="form"
          >
            <textarea
              id="postCaption"
              placeholder="Post message to group"
            />
          </div>
          <button
            className="post-button"
            onClick={[Function]}
            type="submit"
          >
             Post Message 
          </button>
        </div>
        <div
          className="posts"
          id="posts-area"
        />
      </div>
    </div>
    <div
      className="side-navbar"
      id="forMessages"
    >
      <button
        className="messages"
        onClick={[Function]}
        type="submit"
      >
        Messages
      </button>
    </div>
  </div>
</div>
`;

exports[`Test View Group Page UI View group page post error renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, me!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        onClick={[Function]}
        type="submit"
      >
        Analytics
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="invitations"
        onClick={[Function]}
        type="submit"
      >
        Group Invitations and Requests 
      </button>
      <button
        className="photos"
        type="submit"
      >
        Photos
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="info-area"
        id="info-area"
      >
        <div
          className="group-view"
        >
          <div
            className="group-info"
          >
            <ul>
              <li
                id="group-id"
              >
                Group Id: undefined
              </li>
              <li
                id="group-name"
              >
                Group Name: undefined
              </li>
              <li
                id="group-description"
              >
                Group Description: undefined
              </li>
              <li
                id="group-admins"
              >
                Admins: 
              </li>
              <li
                id="is-public"
              >
                Private
              </li>
            </ul>
          </div>
        </div>
        <div
          className="join-group"
          id="join-group-id"
        >
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Leave group"
          />
        </div>
      </div>
       
      <div
        className="view-area"
        id="groups-area"
      >
        <div
          id="revoke"
        >
          <label
            htmlFor="revokeAdmin"
          >
            Revoke Admin: 
            <input
              id="revokeAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add"
        >
          <label
            htmlFor="addAdmin"
          >
            Add Admin: 
            <input
              id="addAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add-non-admin"
        >
          <label
            htmlFor="addNonAdmin"
          >
            Invite a non-Admin member:
            <input
              id="addNonAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
      </div>
      <div
        className="posts-area"
      >
        <div
          className="text-input"
        >
          Create Post:
          <p>
            Error: post is too large
          </p>
          <div
            id="options"
          >
            <label
              className="list"
              htmlFor="textInput"
            >
              <input
                id="textInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Text message
            </label>
            <label
              className="list"
              htmlFor="audioInput"
            >
              <input
                id="audioInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Audio message
            </label>
            <label
              className="list"
              htmlFor="videoInput"
            >
              <input
                id="videoInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Video message
            </label>
            <label
              className="list"
              htmlFor="imageInput"
            >
              <input
                id="imageInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Image message
            </label>
          </div>
          <div
            id="form"
          >
            <textarea
              id="postCaption"
              placeholder="Post message to group"
            />
          </div>
          <button
            className="post-button"
            onClick={[Function]}
            type="submit"
          >
             Post Message 
          </button>
        </div>
        <div
          className="posts"
          id="posts-area"
        />
      </div>
    </div>
    <div
      className="side-navbar"
      id="forMessages"
    >
      <button
        className="messages"
        onClick={[Function]}
        type="submit"
      >
        Messages
      </button>
    </div>
  </div>
</div>
`;

exports[`Test View Group Page UI View group page renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, me!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        onClick={[Function]}
        type="submit"
      >
        Analytics
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="invitations"
        onClick={[Function]}
        type="submit"
      >
        Group Invitations and Requests 
      </button>
      <button
        className="photos"
        type="submit"
      >
        Photos
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="info-area"
        id="info-area"
      >
        <div
          className="group-view"
        >
          <div
            className="group-info"
          >
            <ul>
              <li
                id="group-id"
              >
                Group Id: undefined
              </li>
              <li
                id="group-name"
              >
                Group Name: undefined
              </li>
              <li
                id="group-description"
              >
                Group Description: undefined
              </li>
              <li
                id="group-admins"
              >
                Admins: 
              </li>
              <li
                id="is-public"
              >
                Private
              </li>
            </ul>
          </div>
        </div>
        <div
          className="join-group"
          id="join-group-id"
        >
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Leave group"
          />
        </div>
      </div>
       
      <div
        className="view-area"
        id="groups-area"
      >
        <div
          id="revoke"
        >
          <label
            htmlFor="revokeAdmin"
          >
            Revoke Admin: 
            <input
              id="revokeAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add"
        >
          <label
            htmlFor="addAdmin"
          >
            Add Admin: 
            <input
              id="addAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
        <div
          id="add-non-admin"
        >
          <label
            htmlFor="addNonAdmin"
          >
            Invite a non-Admin member:
            <input
              id="addNonAdmin"
              placeholder="Enter username"
              type="text"
            />
          </label>
          <input
            id="submit"
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>
      </div>
      <div
        className="posts-area"
      >
        <div
          className="text-input"
        >
          Create Post:
          <div
            id="options"
          >
            <label
              className="list"
              htmlFor="textInput"
            >
              <input
                id="textInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Text message
            </label>
            <label
              className="list"
              htmlFor="audioInput"
            >
              <input
                id="audioInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Audio message
            </label>
            <label
              className="list"
              htmlFor="videoInput"
            >
              <input
                id="videoInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Video message
            </label>
            <label
              className="list"
              htmlFor="imageInput"
            >
              <input
                id="imageInput"
                name="msgType"
                onClick={[Function]}
                type="radio"
              />
              Image message
            </label>
          </div>
          <div
            id="form"
          >
            <textarea
              id="postCaption"
              placeholder="Post message to group"
            />
          </div>
          <button
            className="post-button"
            onClick={[Function]}
            type="submit"
          >
             Post Message 
          </button>
        </div>
        <div
          className="posts"
          id="posts-area"
        />
      </div>
    </div>
    <div
      className="side-navbar"
      id="forMessages"
    >
      <button
        className="messages"
        onClick={[Function]}
        type="submit"
      >
        Messages
      </button>
    </div>
  </div>
</div>
`;
